pkg/services/rendering/capabilities.go

```diff
@@ -0,0 +1,54 @@
+package rendering
+
+import (
+	"errors"
+
+	"github.com/Masterminds/semver"
+)
+
+type Capability struct {
+	name             CapabilityName
+	semverConstraint string
+}
+
+type CapabilityName string
+
+const (
+	ScalingDownImages CapabilityName = "ScalingDownImages"
+	FullHeightImages  CapabilityName = "FullHeightImages"
+)
+
+var ErrUnknownCapability = errors.New("unknown capability")
+var ErrInvalidPluginVersion = errors.New("invalid plugin version")
+
+func (rs *RenderingService) HasCapability(capability CapabilityName) (CapabilitySupportRequestResult, error) {
+	var semverConstraint string
+	for i := range rs.capabilities {
+		if rs.capabilities[i].name == capability {
+			semverConstraint = rs.capabilities[i].semverConstraint
+		}
+	}
+
+	if semverConstraint == "" {
+		return CapabilitySupportRequestResult{}, ErrUnknownCapability
+	}
+
+	if !rs.IsAvailable() {
+		return CapabilitySupportRequestResult{IsSupported: false, SemverConstraint: semverConstraint}, ErrRenderUnavailable
+	}
```

Comment:
Do we really need to return the semver constraint here? Else, I'd put that at the beginning of this function to return early if the renderer is unavailable.

Reply:
done!